Sources/DVR/Session.swift

```diff
@@ -160,8 +160,8 @@ open class Session: URLSession {
         return Cassette(dictionary: json)
     }
 
-    func finishTaskAndPersist(_ task: URLSessionTask, interaction: Interaction, playback: Bool) {
-        needsPersistence = (needsPersistence || !playback)
+    func finishTaskWithInteraction(_ task: URLSessionTask, interaction: Interaction, playback: Bool) {
+        needsPersistence = !playback
```

Comment:
I would recommend:
`func finish(task: URLSessionTask, persistingInteraction: Interaction, playback: Bool)`

Reply:
Self is a Session not a URLSessionTask. I don't know if removing it would work. Also it would require to change the logic from DVR that was already there when we started with it. If our plan is to contribute this back I'm not sure if it makes the most sense to change it.